cmd/livepeer/livepeer.go

```diff
@@ -622,13 +624,19 @@ func main() {
 			glog.Infof("Redeemer started on %v", *httpAddr)
 		}
 
-		// If the node address is an on-chain registered address, start the reward service
-		t, err := n.Eth.GetTranscoder(n.Eth.Account().Address)
-		if err != nil {
-			glog.Error(err)
-			return
+		if !isFlagSet["reward"] {
```

Comment:
Can't we just do `if *reward` because it's just a boolean and not an actual value ? Anyway there's little actual difference here so leaving it up to you.

Reply:
If we check the value of `*reward` here, the code in the following block would never be executed because the default value of `*reward` is false even if the user does not explicitly set the `-reward` flag. Instead, we can check `isFlagSet["reward"]` which will be false when the `-reward` flag is not set. The resulting behavior is:

- When the `-reward` flag is not set, if the node's address is registered on-chain, set `*reward` to true and *then* use the value of `*reward` to determine whether to enable/disable the reward service
    - This should preserve the existing behavior that orchestrator operators expect - if the node is run with `-orchestrator` and it uses an on-chain registered address, the node runs a reward service
- When the `-reward` flag is set, use the value of `*reward` to determine whether to enable/disable the reward service

Reply:
AFAIK it is still not different from 

```
if !*reward {
  if on-chain addr == node addr {
   *reward = true
}
}

if *reward {
 // start  reward service 
}
```

Reply:
In your example, the reward service would be enabled when the node's address is registered on-chain even if the user specifies `-reward=false`. In the current code,  the user can disable the reward service even when the node's address is registered on-chain by specifying `-reward=false`.

Reply:
Right, but that would be completely useless at this stage since the on-chain address is the only one that can call reward 

Which means starting a node with the on-chain addr but `-reward=false` would only be useful in the case of key duplication which we don't advise to do anyway.

Reply:
You're right that the use cases for running a node with an on-chain registered address and `-reward=false` feel limited right now. However, I think disabling the reward service when the user specifies `-reward=false` even if the node's address is registered on-chain better matches user expectations. As a user, if I know that `-reward` enables/disables the reward service, I expect `-reward=false` to disable the reward service. If we can better match flag usage with user expectations, I think making a small code change is worthwhile.